hal/armv7a/zynq7000/zynq.c

```diff
@@ -595,32 +595,7 @@ static void _zynq_peripherals(void)
 {
 	_zynq_slcrUnlock();
 
-	/* UART_0 Initialization
-	 * TxD
-	 * TRI_ENABLE = 0; L0_SEL = 0; L1_SEL = 0; L2_SEL = 0; L3_SEL = 7; Speed = 0; IO_Type = 1; PULLUP = 0; DisableRcvr = 0 */
-	*(zynq_common.slcr + slcr_mio_pin_11) = (*(zynq_common.slcr + slcr_mio_pin_11) & ~0x00003fff) | 0x000002e0;
-
-	/* RxD
-	 * TRI_ENABLE = 1; L0_SEL = 0; L1_SEL = 0; L2_SEL = 0; L3_SEL = 7; Speed = 0; IO_Type = 1; PULLUP = 0; DisableRcvr = 0 */
-	*(zynq_common.slcr + slcr_mio_pin_10) = (*(zynq_common.slcr + slcr_mio_pin_10) & ~0x00003fff) | 0x000002e1;
-
-
-	/* UART_1 Initialization
-	 * TxD
-	 * TRI_ENABLE = 0; L0_SEL = 0; L1_SEL = 0; L2_SEL = 0; L3_SEL = 7; Speed = 0; IO_Type = 1; PULLUP = 0; DisableRcvr = 0 */
-	*(zynq_common.slcr + slcr_mio_pin_48) = (*(zynq_common.slcr + slcr_mio_pin_48) & ~0x00003fff) | 0x000002e0;
-
-	/* RxD
-	 * TRI_ENABLE = 1; L0_SEL = 0; L1_SEL = 0; L2_SEL = 0; L3_SEL = 7; Speed = 0; IO_Type = 1; PULLUP = 0; DisableRcvr = 0 */
-	*(zynq_common.slcr + slcr_mio_pin_49) = (*(zynq_common.slcr + slcr_mio_pin_49) & ~0x00003fff) | 0x000002e1;
-
-
-	/* Define UARTs' clocks speed
-	 * IO_PLL / 20 :  1000 MHz / 20 = 50 MHz
-	 * CLKACT0 = 0x0; CLKACT1 = 0x1; SRCSEL = 0x0; DIVISOR = 0x14 */
-	*(zynq_common.slcr + slcr_uart_clk_ctrl) = (*(zynq_common.slcr + slcr_uart_clk_ctrl) & ~0x00003f33) | 0x00001402 | 0x1;
-
-	/* Enable UART_0 & UART_1 clock */
+	/* Enable UART_0 & UART_1 clock simultaneously to avoid race condition */
 	*(zynq_common.slcr + slcr_aper_clk_ctrl) = *(zynq_common.slcr + slcr_aper_clk_ctrl) | (1 << 21) | (1 << 20);
```

Comment:
You can make `_zynq_setAmbaClk` as a part of the interface, similar to the `_zynq_setMIO` and `_zynq_peripherals` would be removed. Then everything related to uart will be done in `console`.

Reply:
I was considering using `_zynq_setAmbaClk` here, however this function sets only one clock at a time. From, what I understood from @agkaminski, we want to set both clocks simultaneously with just one write, so I left it like it is.